src/Core/src/Platform/Windows/NavigationViewExtensions.cs

```diff
@@ -270,5 +272,22 @@ public static void UpdateFlyoutWidth(this MauiNavigationView navigationView, IFl
 			// At some point this Template Setting is going to show up with a bump to winui
 			//handler.PlatformView.OpenPaneLength = handler.PlatformView.TemplateSettings.OpenPaneWidth;
 		}
+
+		public static async Task UpdateFlyoutIconAsync(this AnimatedIcon platformView, IImageSource? imageSource, IImageSourceServiceProvider? provider)
+		{
+			if (platformView is null)
+				return;			
+
+			if (provider is not null && imageSource is not null)
+			{
+				var service = provider.GetRequiredImageSourceService(imageSource);
+				var nativeImageSource = await service.GetImageSourceAsync(imageSource);
+
+				platformView.Source = null;
```

Comment:
What are your thoughts on changing the height/width of the AnimatedIcon? Right now, it forces the width and height to 16 

https://github.com/microsoft/microsoft-ui-xaml/blob/a7183df20367bc0e2b8c825430597a5c1e6871b6/dev/NavigationView/NavigationView_rs1_themeresources.xaml#L380-L381

So, the Icon is super tiny. 

I'm wondering if we should just set the width and height of the `AnimatedIcon` to `Double.Nan`

This sizes the book icon to a bit better of a size.
![image](https://github.com/dotnet/maui/assets/5375137/75335a39-b355-426c-82a7-1a7b351d6471)

It looks like the outer container constrains the icon so even if the user loads in a massive image it'll still shrink.

Here's the .NET MAUI Bot inside the sandbox

![image](https://github.com/dotnet/maui/assets/5375137/4d980036-c32c-4181-9e51-25063615c6ea)

Reply:
The container limits the size, I think that's fine.